src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -326,6 +326,48 @@ public boolean isTableCompatible(final String tableName) {
     return compatible;
   }
 
+  @Override
+  public void appendMetaColIfNotExist(final String tableName) {
+    checkConnection();
+    InternalUtils.assertNotEmpty("tableName", tableName);
+    String query = "desc table identifier(?)";
+    PreparedStatement stmt = null;
+    ResultSet result = null;
+    boolean hasMeta = false;
+    boolean isVariant = false;
+    try {
+      stmt = conn.prepareStatement(query);
+      stmt.setString(1, tableName);
+      result = stmt.executeQuery();
+      while (result.next()) {
+        if (result.getString(1).equals("RECORD_METADATA")) {
+          hasMeta = true;
+          if (result.getString(2).equals("VARIANT")) {
+            isVariant = true;
+          }
+          break;
+        }
+      }
+    } catch (SQLException e) {
+      logError("table {} doesn't exist", tableName);
```

Comment:
Why we only log here instead of thrown the exception and stop the connector?

Reply:
Because this is unlikely to happen. The table should pass the existence checking before a meta column can be appended.

Reply:
I could add an error about this, but it might only happen when the table is deleted immediately after the existence checking.